src/main/java/eu/arrowhead/client/library/ArrowheadService.java

```diff
@@ -453,6 +468,79 @@ public OrchestrationResponseDTO queryOrchestrationStore(final long systemId) {
 		return response.getBody();
 	}
 	
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Make WSS connection with the specified service reachability details.
+	 *
+	 * @param handler WebSocket handler to use for message exchanges
+	 * @param address String value which represents the host where the service is available.
+	 * @param port int value which represents the port where the service is available
+	 * @param serviceUri String value which represents the URI where the service is available.
+	 * @param token (nullable) String value which represents the token for being authorized at the provider side if necessary. Token could be received in orchestration response per interface type.
+	 * @param queryParams (nullable) String... variable arguments which represent the additional key-value http(s) query parameters if any necessary. E.g.: "k1", "v1", "k2", "v2".
+	 * @return A WebSocketConnection manager that is used for all full-duplex communication
+	 *
+	 * @throws InvalidParameterException when service URL can't be assembled.
+	 * @throws CertificateException when a certificate or SSL error occurs
+	 * @throws ArrowheadException when the communication is managed via Gateway Core System and internal server error happened.
+	 */
+	public WebSocketConnectionManager connnectServiceWS(final WebSocketHandler handler, final String address, final int port, final String serviceUri,
+														final String token, final String... queryParams) {
+
+		if (Utilities.isEmpty(address)) {
+			throw new InvalidParameterException("address cannot be null or blank.");
+		}
+		if (Utilities.isEmpty(serviceUri)) {
+			throw new InvalidParameterException("serviceUri cannot be null or blank.");
+		}
+		if (port < 0 || port > 65535){
+			throw new InvalidParameterException("port cannot be negative or larger that 65535");
+		}
+
+		/* Handle query parameters */
+		String[] validatedQueryParams;
+		if (queryParams == null) {
+			validatedQueryParams = new String[0];
+		} else {
+			validatedQueryParams = queryParams;
+		}
+
+		String wsSec = "ws";
```

Comment:
You have wrote the `getUriSchemeWS()`, but not call it at all currently. I suggest to remove these lines, and use your private method, where you need the value.

Reply:
Fixed